File: ui/src/main/messenger-ui/src/app/models/messages/dialogs-previews-response.ts
import {Message} from './message';
import {MessageType} from './message-types';
import {DialogPreview} from './dialog-preview';

export class DialogsPreviewsResponse extends Message {

  dialogPreviews: DialogPreview[];

  get messageType(): MessageType {
    return MessageType.DIALOGS_PREVIEWS_RESPONSE;
  }

}

File: messaging-server/src/main/java/com/freiberg/model/messages/DialogsPreviewsResponse.java
package com.freiberg.model.messages;

import com.freiberg.model.DialogPreview;
import com.freiberg.model.MessageType;
import com.freiberg.model.messages.AbstractMessage;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.Collection;

@EqualsAndHashCode(callSuper = true)
@AllArgsConstructor
@NoArgsConstructor
@Data
public class DialogsPreviewsResponse extends AbstractMessage {

    private Collection<DialogPreview> dialogPreviews;

    @Override
    public MessageType getMessageType() {
        return MessageType.DIALOGS_PREVIEWS_RESPONSE;
    }
}
